task2/src/router/userRouter.ts

```diff
@@ -0,0 +1,75 @@
+import express, { Request, Response } from 'express';
+import { getAutoSuggestUsers } from '../utils/getAutoSuggestUsers';
+import { UserRequestSchema, validator } from '../middleware/validator';
+import { UserValidationSchema } from '../schemas/User';
+import { ValidatedRequest } from 'express-joi-validation';
+import { User } from '../types/User';
+import { v4 as uuid } from 'uuid';
+
+export let users: User[] = [
+    { age: 23, isDeleted: false, login: 'Dmytro', password: 'admin', id: uuid() },
+    { age: 24, isDeleted: false, login: 'Pavlo', password: 'mod', id: uuid() },
+    { age: 27, isDeleted: false, login: 'Oleh', password: 'guest', id: uuid() },
+    { age: 18, isDeleted: false, login: 'Zekora', password: 'guest', id: uuid() },
+    { age: 44, isDeleted: false, login: 'Artem', password: 'guest', id: uuid() },
+    { age: 20, isDeleted: false, login: 'Daniel', password: 'mod', id: uuid() }
+];
+
+export const userRouter = express.Router();
+
+userRouter.param('id', (req: Request, res: Response, next, id) => {
+    const userById = users.find((user) => user.id === id);
+    if (userById) req.user = userById;
+    next();
+});
+
+userRouter.route('/users')
+    .get((req: Request, res: Response) => {
+        let { limit, filter } = req.query;
+        if (!limit) limit = '10';
+        if (!filter) filter = '';
+        const suggestedUsers = getAutoSuggestUsers(String(filter), String(limit));
+
+        res.json(suggestedUsers);
+    })
+    .post(validator.body(UserValidationSchema),
+        (req: ValidatedRequest<UserRequestSchema>, res: Response) => {
+            const user: User = { ...req.body, id: uuid() };
+            users.push(user);
+            res.status(201);
+            res.send(`User with id ${user.id} successfully created`);
+        });
+
+userRouter.route('/users/:id')
+    .get((req: Request, res: Response) => {
```

Comment:
You should notify in case user was not found(400 error code)

Reply:
Is it actually 400 or 404 when user **not found** ?

Reply:
or it depends))

Reply:
done